compiler/cgen.nim

```diff
@@ -560,7 +561,7 @@ proc loadDynamicLib(m: BModule, lib: PLib) =
         n.info = lib.path.info
         appcg(m, loadlib, "($1 = #nimLoadLibrary($2))$n",
               [tmp, genStringLiteral(m, n)])
-      appcg(m, m.s[cfsDynLibInit],
+      appcg(m, m.s[cfsDynLibInitDlopen],
```

Comment:
yes; so that all dlopen code happens before all dlsym code.
the original bug I was hitting was rather puzzling until I realized `loadDynamicLib` was reentrant